Tan_Heng_Yeow/progress.md

```diff
@@ -196,27 +200,80 @@ As mentioned in last week's retrospect, I intend to combine the work done for st
 Week 9 Plan:
 
 4th stage:
-- [ ] Fix any new issues from Week 8
-- [ ] Implement skeleton feedback mechanism for each row in existing students' spreadsheet interface.
+- [X] Fix any new issues from Week 8
+- [X] Implement skeleton feedback mechanism for each row in existing students' spreadsheet interface.
+
+Works:
+- Progress for this week marked at this [checkpoint](https://github.com/TEAMMATES/teammates/pull/8960/commits/055fd77d4b5ef7d736e0a1641f84c8f52f80f119).
+- Worked on this [PR](https://github.com/TEAMMATES/teammates/pull/8954).
+
+Retrospect on Week 9:
+
+I took a short break from the existing project by working on other issues present in TEAMMATES. The reason was that I have been delving into mainly the `Courses` aspect of TEAMMATES and wanted to explore other aspects too. Furthermore, the [issue](https://github.com/TEAMMATES/teammates/issues/8791) was marked as `p.High` and was not actively worked on. By submitting the fix, I got a better understanding of AJAX implementation in TEAMMATES, the `Sessions` aspect and handling of `DateTime` fields. 
+
+The rest of the week was spent exploring on a suitable mechanism for each row in the existing students' spreadsheet interface. The result of the research was that using renderers in `Handsontable` is suitable for the proposed feedback mechanism. This is because custom renderers can be built to suit the needs of this phase of the project by making use of the primitive renderers provided by `Handsontable`. For example, custom renderers allows us to display HTML content in a cell. I plan to use HTML status icons to indicate the type of feedback (success/failure) after backend validation and HTML tooltips as a way to display the content of the feedback. With the introduction of HTML content in the existing students' spreadsheet interface, there might be a risk of XSS attacks. To protect against XSS attacks, I took extra precaution to make sure that back-end sanitization has been put into place for the action I created in the previous stage that processes content sent from the existing students' spreadsheet interface.
```

Comment:
I don't see how introducing HTML poses a risk of XSS. XSS is a risk when we display user content, regardless of whether it is HTML or not. If the HTML added is totally controlled by our app (e.g. error messages without repeating user content) then that would not be a XSS risk. so which case is yours?

Reply:
This was mentioned during the time I introduced status icons in the spreadsheet through the use of Handontable `HTMLRenderer` which allows any HTML content to be displayed in the spreadsheet (for that row/column which uses it). An user can enter `<script>` or other potentially malicious tags using the cell editor in these fields.

The point made refers to the possibility of user sending crafted HTML content to the backend and when left unsanitized would lead to a XSS attack. 

This is not relevant now since I am not using the `HTMLRenderer` anymore but instead using `TextRenderer` for the error messages.